code/game/machinery/vending.dm

```diff
@@ -710,7 +710,11 @@
 	var/mob/living/target = locate() in view(7,src)

```

Comment:
would probably be better to build a temporay list of all living mobs in `view(7, src)`, iterate over it, skip SSDs, then pick, from that list, which one to fire at.

I suspect `locate` has biases in it, anyway.